src/common/util.cpp

```diff
@@ -902,20 +895,26 @@ std::string get_nix_version_display_string()
       return false;
     }
 
-    // resolve to IP
-    boost::asio::io_service io_service;
-    boost::asio::ip::tcp::resolver resolver(io_service);
-    boost::asio::ip::tcp::resolver::query query(u_c.host, "");
-    boost::asio::ip::tcp::resolver::iterator i = resolver.resolve(query);
-    while (i != boost::asio::ip::tcp::resolver::iterator())
+    if (u_c.host == "localhost" || boost::ends_with(u_c.host, ".localhost")) { // RFC 6761 (6.3)
+      MDEBUG("Address '" << address << "' is local");
+      return true;
+    }
+
+    boost::asio::ip::address parsed_ip;
+    try
     {
-      const boost::asio::ip::tcp::endpoint &ep = *i;
-      if (ep.address().is_loopback())
-      {
-        MDEBUG("Address '" << address << "' is local");
-        return true;
-      }
-      ++i;
+      parsed_ip = boost::asio::ip::address::from_string(u_c.host);
```

Comment:
Use the overload that takes a `boost::system::error_code`, `parsed_ip` can be made `const`, etc.

Reply:
The error message might as well contain `error.message()` in the `MWARNING` output too.

Reply:
Thanks